fpgaconvnet/models/partition/update.py

```diff
@@ -40,6 +40,21 @@ def update(self):
         if self.graph.nodes[node]["type"] in MULTIPORT_LAYERS_IN:
             self.update_multiport_buffer_depth(node)
 
+def update_io_ports(self):
+    input_nodes = graphs.get_input_nodes(self.graph)
+    for node_in in input_nodes:
+        if self.graph.nodes[node_in]["type"] in MULTIPORT_LAYERS_IN:
+            self.ports_in += self.graph.nodes[node_in]["hw"].ports_in - 1
+    self.size_in     *= self.ports_in
```

Comment:
what if ports_in is reduced

Reply:
This one is going to be removed after all.
I know it had issues like the one you pointed out but I wasn't actually using that